python_benchmarks/qr_codes/qr.py

```diff
@@ -106,65 +214,108 @@ def get_gold_corners(label_path):
-            one_corners = np.roll(one_corners, 1, 0)
-            dist = min(dist, get_norm(gold_corner, one_corners, type_dist))
-        if dist < prev_dist:
-            best_id = cur_id
-        cur_id += 1
-
-    if type_dist is TypeNorm.intersection_over_union:
-        rect = cv.boundingRect(np.concatenate((gold_corner, corners[best_id]), dtype=np.float32))
-        mask1 = np.zeros((rect[2], rect[3]), dtype=np.uint8)
-        mask1 = cv.fillConvexPoly(mask1, gold_corner.astype(np.int32) - (rect[0], rect[1]), 255)
-        mask2 = np.zeros((rect[2], rect[3]), dtype=np.uint8)
-        mask2 = cv.fillConvexPoly(mask2, corners[best_id].astype(np.int32) - (rect[0], rect[1]), 255)
-        a = cv.countNonZero(cv.bitwise_and(mask2, mask1))
-        b = max(cv.countNonZero(cv.bitwise_or(mask2, mask1)), 1)
-        dist = 1. - a / b
-    return dist, best_id
+def get_time():
+    return datetime.datetime.now().strftime("%Y_%m_%d-%I_%M_%S_%p")
+
+
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return json.JSONEncoder.default(self, obj)
+
+
+def set_plt():
+    # Turn interactive plotting off
+    plt.ioff()
+    plt.rcParams["figure.figsize"] = (15, 9)
+    plt.rcParams["figure.subplot.bottom"] = 0.3
+    plt.rcParams["figure.subplot.left"] = 0.05
+    plt.rcParams["figure.subplot.right"] = 0.99
+
+
+def get_and_print_category_statistic(obj_type, category, statistics, accuracy_threshold, path):
+    if obj_type == "decode":
+        decoded_info = list(deepflatten(statistics, ignore=str))
+        decoded = sum(1 for info in decoded_info if info != "")
+        category_statistic = OrderedDict(
+            [("category", category), ("decoded", decoded/len(decoded_info)), ("total decoded", decoded)])
+        return category_statistic
+    objs = np.array(list(deepflatten(statistics)))
+    detected = objs[objs < accuracy_threshold]
+    category_statistic = OrderedDict(
+        [("category", category), ("detected " + obj_type, len(detected) / max(1, len(objs))),
+         ("total detected " + obj_type, len(detected)), ("total " + obj_type, len(objs)),
+         ("average detected error " + obj_type, np.mean(detected))])
+    data_frame = pd.DataFrame(objs)
+    data_frame.hist(bins=500)
+    plt.title(category + ' ' + obj_type)
+    plt.xlabel('error')
+    plt.xticks(np.arange(0., float(accuracy_threshold) + .25, .25))
+    plt.ylabel('frequency')
+    # plt.show()
+    plt.savefig(path + '/' + category + '_' + obj_type + '.jpg')
+    plt.close()
+    return category_statistic
+
+
+def print_statistics(filename, distances, accuracy_threshold, output_path, per_image_statistic):
+    output_dict = output_path + '/' + filename
+    if not os.path.exists(output_dict):
+        os.mkdir(output_dict)
+    with open(output_dict + "/" + "distances" + '.json', 'w') as fp:
+        json.dump(distances, fp, cls=NumpyEncoder)
```

Comment:
please use `indent=4` for "human readable" json. Otherwise it's single line document.

Reply:
added